[Bug]: DSWx-S1 unit tests fail due to missing hdf5 package

### Checked for duplicates

Yes - I've already checked

### Describe the bug

The unit test suite for the DSWx-S1 PGE is currently failing in Jenkins with the following error:

```
_______ ERROR collecting src/opera/test/pge/dswx_s1/test_dswx_s1_pge.py ________
ImportError while importing test module '/workspace/src/opera/test/pge/dswx_s1/test_dswx_s1_pge.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/conda/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
../src/opera/test/pge/dswx_s1/test_dswx_s1_pge.py:24: in <module>
    from opera.pge.dswx_s1.dswx_s1_pge import DSWxS1Executor
/home/conda/opera/pge/dswx_s1/dswx_s1_pge.py:22: in <module>
    from opera.util.metadata_utils import get_geographic_boundaries_from_mgrs_tile
/home/conda/opera/util/metadata_utils.py:14: in <module>
    import h5py
/opt/conda/lib/python3.9/site-packages/h5py/__init__.py:33: in <module>
    from . import version
/opt/conda/lib/python3.9/site-packages/h5py/version.py:15: in <module>
    from . import h5 as _h5
h5py/h5.pyx:1: in init h5py.h5
    ???
E   ImportError: /opt/conda/lib/python3.9/site-packages/h5py/defs.cpython-39-x86_64-linux-gnu.so: undefined symbol: H5Pset_fapl_ros3
```

This is likely related to the recent changes to the build framework for the R2 PGEs, where hdf5 was removed from requirements.txt, since pip does not recognize the module name (but conda does). The build script for DSWx-S1 needs to be updated to install the hdf5 package via conda to ensure the symbol above is available.

### What did you expect?

The unit test suite for DSWx-S1 passes completely.

### Reproducible steps

```bash
Run the build/test pipeline for any branch and observe the failure in the pytest log for DSWx-S1.
```


### Environment

```bash
- Version of this software [e.g. vX.Y.Z]
- Operating System: [e.g. MacOSX with Docker Desktop vX.Y]
...
```

--- a/.ci/docker/Dockerfile_dswx_s1
+++ b/.ci/docker/Dockerfile_dswx_s1
@@ -50,7 +50,8 @@ RUN set -ex \
     && cp ${PGE_DEST_DIR}/opera/scripts/*_entrypoint.sh ${CONDA_ROOT}/bin \
     && chmod +x ${CONDA_ROOT}/bin/*_entrypoint.sh \
     && . ${CONDA_ROOT}/bin/activate root \
-    && conda install --yes --channel conda-forge --file ${PGE_DEST_DIR}/opera/requirements.txt
+    && python -m pip install -r ${PGE_DEST_DIR}/opera/requirements.txt \
+    && conda install --yes --channel conda-forge hdf5
 
 # Set the Docker entrypoint and clear the default command
 ENTRYPOINT ["sh", "-c", "exec ${CONDA_ROOT}/bin/pge_docker_entrypoint.sh \"${@}\"", "--"]

--- a/.ci/scripts/build_dswx_s1.sh
+++ b/.ci/scripts/build_dswx_s1.sh
@@ -78,7 +78,7 @@ else
 fi
 
 # Build the PGE docker image
-docker build ${PLATFORM} --rm --force-rm -t ${IMAGE}:${TAG} \
+docker build ${PLATFORM} --progress plain --rm --force-rm -t ${IMAGE}:${TAG} \
     --build-arg SAS_IMAGE=${SAS_IMAGE} \
     --build-arg BUILD_DATE_TIME=${BUILD_DATE_TIME} \
     --build-arg BUILD_VERSION=${TAG} \

